metis-validation/metis-validation-rest/src/main/java/eu/europeana/validation/rest/ValidationController.java

```diff
@@ -112,28 +112,28 @@ public ValidationResultList batchValidate(
 
     final List<ByteArrayInputStream> records;
     try {
-      records = new ZipFileReader().getContentFromZipFile(providedZipFile.getInputStream());
+      records = new CompressedFileHandler().getContentFromZipFile(providedZipFile.getInputStream());
     } catch (IOException e) {
       throw new ServerException(e);
     }
     if (records.isEmpty()) {
       throw new ServerException("No suitable records found in zip file.");
     }
-        
+
     try {
-      ValidationResultList list = validator.batchValidation(targetSchema, null, null, records);
-      // TODO JV Note the condition below means that list.success is ALWAYS true (or a NPE occurs). This is probably not the purpose here.
-      if (list.getResultList() != null || list.getResultList().isEmpty()) {
-        list.setSuccess(true);
-      }
-      if (list.isSuccess()) {
-        return list;
-      } else {
-        throw new BatchValidationException("Batch service failed", list);
+      ValidationResultList validationResultList =
+          validator.batchValidation(targetSchema, null, null, records);
+
+      if (validationResultList != null) {
+        if (validationResultList.getResultList() != null || validationResultList.getResultList().isEmpty()) {
+          validationResultList.setSuccess(true);
+          return validationResultList;
+        }
+        throw new BatchValidationException("Batch validation service failed.", validationResultList);
       }
-    } catch (ExecutionException e) {
-      throw new ServerException(e);
-    } catch (InterruptedException e) {
+      return null;
```

Comment:
I think this ```return null``` goes at the end. what do you think?
```
@PostMapping(value = SCHEMA_BATCH_VALIDATE)
  @ResponseBody
  @ApiOperation(value = "Validate zip file based on schema", response = ValidationResultList.class)
  public ValidationResultList batchValidate(
      @ApiParam(value = "schema") @PathVariable("schema") String targetSchema,
      @ApiParam(value = "file") @RequestPart("file") MultipartFile providedZipFile)
      throws ServerException, BatchValidationException {

    if (!schemaProvider.isPredefined(targetSchema)) {
      throw new BatchValidationException("It is not predefined schema.",
          new ValidationResultList());
    }

    final List<ByteArrayInputStream> records;
    try {
      records = new CompressedFileHandler().getContentFromZipFile(providedZipFile.getInputStream());
    } catch (IOException e) {
      throw new ServerException(e);
    }
    if (records.isEmpty()) {
      throw new ServerException("No suitable records found in zip file.");
    }

    try {
      ValidationResultList validationResultList =
          validator.batchValidation(targetSchema, null, null, records);

      if (validationResultList != null) {
        if (validationResultList.getResultList() != null || validationResultList.getResultList().isEmpty()) {
          validationResultList.setSuccess(true);
          return validationResultList;
        }
        throw new BatchValidationException("Batch validation service failed.", validationResultList);
      }
    } catch (ExecutionException | InterruptedException e) {
      Thread.currentThread().interrupt();
      throw new ServerException(e);
    }
    return null;
  }
```

Reply:
Of course it goes ;)